contracts/peg/BasePSM.sol

```diff
@@ -0,0 +1,451 @@
+    function getRedeemAmountOut(uint256 amountVoltIn)
+        public
+        view
+        override
+        returns (uint256 amountTokenOut)
+    {
+        amountTokenOut = _getRedeemAmountOut(amountVoltIn);
+    }
+
+    /// @notice the maximum mint amount out
+    function getMaxMintAmountOut() external view override returns (uint256) {
+        return volt().balanceOf(address(this));
+    }
+
+    /// @notice a flag for whether the current balance is above (true) or below (false) the reservesThreshold
+    function hasSurplus() external view override returns (bool) {
+        return balance() > reservesThreshold;
+    }
+
+    /// @notice an integer representing the positive surplus or negative deficit of contract balance vs reservesThreshold
+    function reservesSurplus() public view override returns (int256) {
+        return balance().toInt256() - reservesThreshold.toInt256();
+    }
+
+    /// @notice function from PCVDeposit that must be overriden
+    function balance() public view virtual override returns (uint256) {
+        return underlyingToken.balanceOf(address(this));
+    }
+
+    /// @notice returns address of token this contracts balance is reported in
+    function balanceReportedIn() public view override returns (address) {
+        return address(underlyingToken);
+    }
+
+    /// @notice override default behavior of not checking VOLT balance
+    function resistantBalanceAndVolt()
+        public
+        view
+        override
+        returns (uint256, uint256)
+    {
+        return (balance(), voltBalance());
+    }
+
+    // ----------- Internal Methods -----------
+
+    /// @notice helper function to get mint amount out based on current market prices
+    /// @dev will revert if price is outside of bounds and bounded PSM is being used
+    function _getMintAmountOut(uint256 amountIn)
+        internal
+        view
+        virtual
+        returns (uint256 amountVoltOut)
+    {
+        Decimal.D256 memory price = readOracle();
+        _validatePriceRange(price);
+
+        amountVoltOut = amountIn / price.value;
+    }
+
+    /// @notice helper function to get redeem amount out based on current market prices
+    /// @dev will revert if price is outside of bounds and bounded PSM is being used
+    function _getRedeemAmountOut(uint256 amountVoltIn)
+        internal
+        view
+        virtual
+        returns (uint256 amountTokenOut)
+    {
+        Decimal.D256 memory price = readOracle();
+        _validatePriceRange(price);
+
+        amountTokenOut = amountVoltIn * price.value;
+    }
+
+    /// @notice Allocates a portion of escrowed PCV to a target PCV deposit
+    function _allocate(uint256 amount) internal virtual {
+        _transfer(address(surplusTarget), amount);
+
+        surplusTarget.deposit();
+
+        emit AllocateSurplus(msg.sender, amount);
+    }
+
+    /// @notice transfer ERC20 token
+    function _transfer(address to, uint256 amount) internal {
```

Comment:
let's just inline this instead of making an additional function call internally.